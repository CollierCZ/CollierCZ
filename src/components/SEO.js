import { graphql, StaticQuery } from "gatsby"
import React from "react";
import Helmet from "react-helmet";
import PropTypes from 'prop-types';

export const PureSEO = ({ basicInfo, articleProps }) => {
  const articleNode = articleProps ? articleProps.articleNode : null;
  const articlePath = articleProps ? articleProps.articlePath : null;
  const siteTitle = basicInfo.kontentItemHome.elements.title.value;
  const title = articleNode ? articleNode.elements.title.value : '';
  const description = articleNode ? articleNode.elements.metadata__description.value : basicInfo.kontentItemHome.elements.metadata__description.value;
  const image = articleNode ? articleNode.elements.teaser.value[0].url : basicInfo.kontentItemHome.elements.splash_image.value[0].url;
  const blogURL = basicInfo.kontentItemHome.elements.base_url.value;
  const logo = basicInfo.kontentItemHome.elements.blog_logo.value[0].url;
  const articleURL = articleNode ? blogURL+"/articles/"+articlePath+"/" : null;

  var schemaOrgJSONLD = [
    {
      "@context": "http://schema.org",
      "@type": "WebSite",
      url: blogURL,
      name: title,
      alternateName: ""
    }
  ];
  if (articleNode) {
    schemaOrgJSONLD.push(
      {
        "@context": "http://schema.org",
        "@type": "BreadcrumbList",
        itemListElement: [
          {
            "@type": "ListItem",
            position: 1,
            item: {
              "@id": articleURL,
              name: title,
              image
            }
          }
        ]
      },
      {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        url: blogURL,
        name: title,
        alternateName: "",
        headline: title,
        image: {
          "@type": "ImageObject",
          url: image
        },
        description
      }
    );
  }
  return (
<Helmet defaultTitle={siteTitle} titleTemplate={"%s | " + siteTitle}
        link={[
          { rel: 'shortcut icon', type: 'image/png', href: `${logo}` }
      ]}
      >
        {/* General tags */}
        <title lang="en">{title}</title>
        <meta name="description" content={description} />
        <meta name="image" content={image} />

        {/* Schema.org tags */}
        <script type="application/ld+json">
          {JSON.stringify(schemaOrgJSONLD)}
        </script>

        <meta property="og:url" content={articleURL ? articleURL : blogURL} />
        {articleURL ? <meta property="og:type" content="article" /> : null}
        <meta property="og:title" content={title} />
        <meta property="og:description" content={description} />
        <meta property="og:image" content={image} />
        {/* no Facebook app
          <meta
            property="fb:app_id"
            content={config.siteFBAppID ? config.siteFBAppID : ""}
          />
        */}

        <meta name="twitter:card" content="summary_large_image" />
         {/* no Twitter creator
          <meta
            name="twitter:creator"
            content={config.userTwitter ? config.userTwitter : ""}
          />
         */}
        <meta name="twitter:title" content={title} />
        <meta name="twitter:description" content={description} />
        <meta name="twitter:image" content={image} />
      </Helmet>
  )
}

export const SEO = props => (
  <StaticQuery
  query={graphql`
    query {
      kontentItemHome{
        ...MetadataFragment
      }
    }
  `}
  render = {data => 
   <PureSEO  articleProps={props} basicInfo={data} />
  }
  />
)

SEO.propTypes = {
  articlePath: PropTypes.string,
  articleNode: PropTypes.object
};

export default SEO;