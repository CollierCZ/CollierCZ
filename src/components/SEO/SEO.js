import React, { Component } from "react";
import Helmet from "react-helmet";
import urljoin from "url-join";



class SEO extends Component {
  render() {
    const config = this.props.seoConfig;
    const { articleNode, articlePath, articleSEO } = this.props;
    let title;
    let description;
    let image;
    let articleURL;
    if (articleSEO) {
      ({ title } = articleNode.title.value);
      description = articleNode.metadata__description;
      image = articleNode.teaser.value[0].url;
      articleURL = urljoin(config.base_url.value, articlePath);
    } else {
      title = config.title.value;
      description = config.metadata__description.value;
      image = config.blog_logo.value[0].url;
    }

    image = urljoin(config.base_url.value, image);
    const blogURL = config.base_url.value;
    const schemaOrgJSONLD = [
      {
        "@context": "http://schema.org",
        "@type": "WebSite",
        url: blogURL,
        name: title,
        alternateName: config.siteTitleAlt ? config.siteTitleAlt : ""
      }
    ];
    if (articleSEO) {
      schemaOrgJSONLD.push(
        {
          "@context": "http://schema.org",
          "@type": "BreadcrumbList",
          itemListElement: [
            {
              "@type": "ListItem",
              position: 1,
              item: {
                "@id": articleURL,
                name: title,
                image
              }
            }
          ]
        },
        {
          "@context": "http://schema.org",
          "@type": "BlogPosting",
          url: blogURL,
          name: title,
          alternateName: config.siteTitleAlt ? config.siteTitleAlt : "",
          headline: title,
          image: {
            "@type": "ImageObject",
            url: image
          },
          description
        }
      );
    }
    return (
      <Helmet>
        {/* General tags */}
        <meta name="description" content={description} />
        <meta name="image" content={image} />

        {/* Schema.org tags */}
        <script type="application/ld+json">
          {JSON.stringify(schemaOrgJSONLD)}
        </script>

        {/* OpenGraph tags */}
        <meta property="og:url" content={articleSEO ? articleURL : blogURL} />
        {articleSEO ? <meta property="og:type" content="article" /> : null}
        <meta property="og:title" content={title} />
        <meta property="og:description" content={description} />
        <meta property="og:image" content={image} />
        <meta
          property="fb:app_id"
          content={config.siteFBAppID ? config.siteFBAppID : ""}
        />

        {/* Twitter Card tags */}
        <meta name="twitter:card" content="summary_large_image" />
        <meta
          name="twitter:creator"
          content={config.userTwitter ? config.userTwitter : ""}
        />
        <meta name="twitter:title" content={title} />
        <meta name="twitter:description" content={description} />
        <meta name="twitter:image" content={image} />
      </Helmet>
    );
  }
}

export default SEO;