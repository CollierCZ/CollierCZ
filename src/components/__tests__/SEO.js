import React from "react"
import { shallow } from "enzyme"
import { PureSEO as SEO } from "../SEO"
const homeQuery = {
  kenticoCloudItemHome: {
      elements: {
        title: {
          value: "Aaron Collier"
        },
        splash_image: {
          assets: [
            {
              url: "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/259d5f15-d354-4701-a299-91cbb7cdf8ce/the-trees-in-the-fall-2052252_640.jpg"
            }
          ]
        },
        blog_logo: {
          assets: [
            {
              url: "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/8af6ebfc-b92f-4603-90f5-5d3dda5d8c3d/quixote-small.jpg"
            }
          ]
        },
        metadata__description: {
          value: "Articles and interesting things."
        },
        socialmedia: {
          value: "mailto:aaron@collier.cz,https://github.com/CollierCZ,https://collier.cz/rss.xml"
        },
        base_url: {
          value: "https://collier.cz"
        }
      }
    }
}

const articleProps= {
  articlePath: "home",
  articleNode: {
    elements: {
      title: {
        value: "Home title"
      },
      metadata__description: {
        value: "An interesting description"
      },
      teaser: {
        value:[
          {
            url: "https://example.com"
          }
        ]
      }
    }
  }
}

describe("SEO", () => {
  it("renders correctly with no article data", () => {
    const component = shallow (<SEO basicInfo={homeQuery} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with article data", () => {
    const component = shallow (<SEO basicInfo={homeQuery} articleProps={articleProps} />)
    expect(component).toMatchSnapshot()
  })
})