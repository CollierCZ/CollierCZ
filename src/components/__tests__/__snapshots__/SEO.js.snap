// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly with article data 1`] = `
<HelmetWrapper
  defaultTitle="Aaron Collier"
  defer={true}
  encodeSpecialCharacters={true}
  link={
    Array [
      Object {
        "href": "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/8af6ebfc-b92f-4603-90f5-5d3dda5d8c3d/quixote-small.jpg",
        "rel": "shortcut icon",
        "type": "image/png",
      },
    ]
  }
  titleTemplate="%s | Aaron Collier"
>
  <title
    lang="en"
  >
    Home title
  </title>
  <meta
    content="An interesting description"
    name="description"
  />
  <meta
    content="https://example.com"
    name="image"
  />
  <script
    type="application/ld+json"
  >
    [{"@context":"http://schema.org","@type":"WebSite","url":"https://collier.cz","name":"Home title","alternateName":""},{"@context":"http://schema.org","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"item":{"@id":"https://collier.cz/articles/home/","name":"Home title","image":"https://example.com"}}]},{"@context":"http://schema.org","@type":"BlogPosting","url":"https://collier.cz","name":"Home title","alternateName":"","headline":"Home title","image":{"@type":"ImageObject","url":"https://example.com"},"description":"An interesting description"}]
  </script>
  <meta
    content="https://collier.cz/articles/home/"
    property="og:url"
  />
  <meta
    content="article"
    property="og:type"
  />
  <meta
    content="Home title"
    property="og:title"
  />
  <meta
    content="An interesting description"
    property="og:description"
  />
  <meta
    content="https://example.com"
    property="og:image"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content="Home title"
    name="twitter:title"
  />
  <meta
    content="An interesting description"
    name="twitter:description"
  />
  <meta
    content="https://example.com"
    name="twitter:image"
  />
</HelmetWrapper>
`;

exports[`SEO renders correctly with no article data 1`] = `
<HelmetWrapper
  defaultTitle="Aaron Collier"
  defer={true}
  encodeSpecialCharacters={true}
  link={
    Array [
      Object {
        "href": "https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/8af6ebfc-b92f-4603-90f5-5d3dda5d8c3d/quixote-small.jpg",
        "rel": "shortcut icon",
        "type": "image/png",
      },
    ]
  }
  titleTemplate="%s | Aaron Collier"
>
  <title
    lang="en"
  />
  <meta
    content="Articles and interesting things."
    name="description"
  />
  <meta
    content="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/259d5f15-d354-4701-a299-91cbb7cdf8ce/the-trees-in-the-fall-2052252_640.jpg"
    name="image"
  />
  <script
    type="application/ld+json"
  >
    [{"@context":"http://schema.org","@type":"WebSite","url":"https://collier.cz","name":"","alternateName":""}]
  </script>
  <meta
    content="https://collier.cz"
    property="og:url"
  />
  <meta
    content=""
    property="og:title"
  />
  <meta
    content="Articles and interesting things."
    property="og:description"
  />
  <meta
    content="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/259d5f15-d354-4701-a299-91cbb7cdf8ce/the-trees-in-the-fall-2052252_640.jpg"
    property="og:image"
  />
  <meta
    content="summary_large_image"
    name="twitter:card"
  />
  <meta
    content=""
    name="twitter:title"
  />
  <meta
    content="Articles and interesting things."
    name="twitter:description"
  />
  <meta
    content="https://assets-us-01.kc-usercontent.com:443/3fcf700a-30e8-4d1d-9e64-43193a89fe7a/259d5f15-d354-4701-a299-91cbb7cdf8ce/the-trees-in-the-fall-2052252_640.jpg"
    name="twitter:image"
  />
</HelmetWrapper>
`;
